llvm/lib/Target/Xtensa/XtensaInstrInfo.cpp

```diff
@@ -185,3 +185,301 @@ void XtensaInstrInfo::loadImmediate(MachineBasicBlock &MBB,
     report_fatal_error("Unsupported load immediate value");
   }
 }
+
+bool XtensaInstrInfo::reverseBranchCondition(
+    SmallVectorImpl<MachineOperand> &Cond) const {
+  assert(Cond.size() <= 4 && "Invalid branch condition!");
+
+  switch (Cond[0].getImm()) {
```

Comment:
Not sure all of these are tested?

Reply:
I tested most of cases for reverseBranchCondition and it seems it works correctly.